bundle/edu.gemini.itc.shared/src/main/java/edu/gemini/itc/shared/ConfigCreator.java

```diff
@@ -0,0 +1,201 @@
+    public static final ItemKey ObserveTypeKey = new ItemKey("observe:observeType");
+    public static final ItemKey ObserveClassKey = new ItemKey("observe:class");
+    public static final ItemKey ExposureTimeKey = new ItemKey("observe:exposureTime");
+    public static final ItemKey CoaddsKey = new ItemKey("observe:coadds");
+    public static final ItemKey InstInstrumentKey = new ItemKey("instrument:instrument");
+    public static final ItemKey ReadModeKey = new ItemKey("instrument:readMode");
+    public static final ItemKey AmpReadModeKey = new ItemKey("instrument:ampReadMode");
+    public static final ItemKey SlitWidthKey = new ItemKey("instrument:slitWidth");
+    public static final ItemKey DisperserKey = new ItemKey("instrument:disperser");
+    public static final ItemKey BuiltinROIKey = new ItemKey("instrument:builtinROI");
+    public static final ItemKey FPUKey = new ItemKey("instrument:fpu");
+    public static final ItemKey CcdXBinning = new ItemKey("instrument:ccdXBinning");
+    public static final ItemKey CcdYBinning = new ItemKey("instrument:ccdYBinning");
+    public static final ItemKey AmpGain = new ItemKey("instrument:gainChoice");
+    public static final ItemKey AmpCount = new ItemKey("instrument:ampCount");
+    public static final ItemKey DetectorManufacturerKey = new ItemKey("instrument:detectorManufacturer");
+    public static final ItemKey GuideWithPWFS2Key = new ItemKey("telescope:guideWithPWFS2");
+    public static final ItemKey TelescopeQKey = new ItemKey("telescope:q");
+    public static final ItemKey TelescopePKey = new ItemKey("telescope:p");
+    public static final ItemKey AOGuideStarTypeKey = new ItemKey("adaptive optics:guideStarType");
+    public static final ItemKey AOSystemKey = new ItemKey("adaptive optics:aoSystem");
+    public static final ItemKey AOFieldLensKey = new ItemKey("adaptive optics:fieldLens");
+
+    private final ItcParameters itcParams;
+    private final ObservationDetails obsDetailParams;
+
+    public ConfigCreator(final ItcParameters p) {
+        this.itcParams = p;
+        this.obsDetailParams = p.observation();
+    }
+
+    // create part of config common for all instruments
+    public Config[] createCommonConfig(int numberExposures) {
+        final CalculationMethod calcMethod = obsDetailParams.calculationMethod();
+        int numberCoadds = calcMethod.coaddsOrElse(1);
+        Config[] dc = new DefaultConfig[numberExposures];
+        String offset = Double.toString(calcMethod.offset());
+        // for spectroscopic observations we consider ABBA offset pattern
+        List<String> spectroscopyOffsets = new ArrayList<>(Arrays.asList("0", offset, offset, "0"));
+        // for imaging observations we consider ABAB offset pattern
+        List<String> imagingOffsets = new ArrayList<>(Arrays.asList("0", offset, "0", offset));
+        List<String> offsetList = new ArrayList<>();
+
+        for (int i = 0; i < (1 + numberExposures / 4); i++) {
```

Comment:
Are we certain that `numberExposures` will always be a multiple of 4? Do we really want multiple copies of either `spectroscopyOffsets` or `imagingOffsets` added to `offsetList`?

Reply:
You are right, this is a horrible way of doing this. I will try to come up with something better.